pkg/secrets/flags.go

```diff
@@ -38,11 +42,16 @@ func (o *CLIOptions) Validate() []error {
 	} else if o.BwUser != "" {
 		credentialsProviderConfigured = append(credentialsProviderConfigured, "bitwarden")
 	}
-
-	if vals := sets.NewString(o.VaultAddr, o.VaultTokenFile, o.VaultPrefix); len(vals) != 1 && ((len(vals) != 3) || vals.Has("")) {
-		errs = append(errs, fmt.Errorf("--vault-addr, --vault-token and --vault-prefix must be specified together"))
-	} else if len(vals) == 3 {
+	b2i := func(b bool) int {
+		if b {
+			return 1
+		}
+		return 0
+	}
+	if n := b2i(o.VaultAddr != "") + b2i(o.VaultToken != "" || o.VaultTokenFile != "") + b2i(o.VaultPrefix != ""); n == 3 {
```

Comment:
```suggestion
	if o.VaultAddr != "" || o.VaultTokenFile != "" || o.VaultPrefix != "" {
	    if o.VaultAddr == "" || o.vaultToken == "" || o.VaultPrefix == "" {
	        return err
	    } else { credentialsProvidersConfigured {...}
```

My point was not that your change is incorrect, but that it is unreadable

Reply:
@bbguimaraes I've included your commit and a fixup into https://github.com/openshift/ci-tools/pull/1857 because it would otherwise conflict with this PR